kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ReverseAddRemoveToAddColumnarBatchIterator.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.internal.replay;
+
+import io.delta.kernel.data.*;
+import io.delta.kernel.internal.actions.AddFile;
+import io.delta.kernel.internal.actions.DeletionVectorDescriptor;
+import io.delta.kernel.internal.actions.RemoveFile;
+import io.delta.kernel.internal.util.Logging;
+import io.delta.kernel.types.BooleanType;
+import io.delta.kernel.types.DataType;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.kernel.utils.Tuple2;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.*;
+
+/**
+ * This class takes an iterator of ({@link FileDataReadResult}, isFromCheckpoint), where the
+ * columnar data inside the FileDataReadResult represents {@link LogReplay#ADD_REMOVE_READ_SCHEMA},
+ * and produces an iterator of {@link DataReadResult} with schema {@link AddFile#READ_SCHEMA}, and
+ * with a selection vector indicating which AddFiles are still active in the table (have not been
+ * tombstoned).
+ */
+public class ReverseAddRemoveToAddColumnarBatchIterator
```

Comment:
We dont have to keep adding "Reverse" in the name everywhere. there is no alternative to differentiate from.